lib/src/components/NxDrawer/__tests__/NxDrawer.test.tsx

```diff
@@ -0,0 +1,283 @@
+    const drawer = getDrawer({ children: <div data-testid="foo"/> })!;
+    expect(within(drawer).getByTestId('foo')).toBe(drawer.children[0]);
+  });
+
+  it('merges any passed in className to the nx-drawer dialog', function() {
+    const drawerWithAddedClassName = getDrawer({ className: 'foo' });
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const drawer = getDrawer()!;
+
+    expect(drawerWithAddedClassName).toHaveClass('foo');
+
+    for (const cls of Array.from(drawer.classList)) {
+      expect(drawerWithAddedClassName).toHaveClass(cls);
+    }
+  });
+
+  it('includes any passed in attributes to the dialog element', function() {
+    const drawer = getDrawer({ id: 'drawer-id', lang: 'en_US' });
+    expect(drawer).toHaveAttribute('id', 'drawer-id');
+    expect(drawer).toHaveAttribute('lang', 'en_US');
+  });
+
+  it('adds the nx-drawer--narrow class when the narrow variant prop is specified', function() {
+    const drawer = getDrawer({ variant: 'narrow' });
+    expect(drawer).toHaveClass('nx-drawer--narrow');
+  });
+
+  describe('NxDrawer event listener support', () => {
+    it('executes onClose callback when ESC key is pressed', function() {
+      const mockOnClose = jest.fn();
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      getDrawer({ onClose: mockOnClose })!;
+
+      const dialog = screen.getByRole('dialog', { hidden: true });
+      expect(mockOnClose).not.toHaveBeenCalled();
+
+      act(() => {
+        fireEvent.keyDown(dialog, { key: 'Escape' });
+      });
+
+      expect(mockOnClose).toHaveBeenCalled();
+    });
+
+    it('executes onClose ONLY when pressing ESC key', async function () {
+      const user = userEvent.setup();
+
+      const mockOnClose = jest.fn();
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const drawer = getDrawer({ onClose: mockOnClose })!;
+
+      await user.keyboard('Tab');
+      await user.keyboard('Enter');
+      await user.keyboard('q');
+      await user.keyboard('Q');
+
+      fireEvent.animationEnd(drawer);
+
+      expect(mockOnClose).not.toHaveBeenCalled();
+    });
+
+    it('executes onCancel when drawer is closed and animation is completed', async function() {
+      const user = userEvent.setup();
+      const mockOnCancel = jest.fn();
+
+      const Fixture = () => {
+        const [open, toggleDrawer] = useToggle(true);
+        return (
+          <NxDrawer open={open} onClose={toggleDrawer} onCancel={mockOnCancel}>
+            <NxDrawer.Header>
+              <NxDrawer.HeaderTitle>Title</NxDrawer.HeaderTitle>
+            </NxDrawer.Header>
+          </NxDrawer>
+        );
+      };
+
+      render(<Fixture />);
+
+      const closeButton = screen.getByRole('button', { name: 'Close', hidden: true });
+      const dialog = screen.getByRole('dialog', { hidden: true });
+
+      await act(async () => {
```

Comment:
Is this `act` necessary? The test seems to pass without it

Reply:
> Is this `act` necessary? The test seems to pass without it

It didn't pass for me for some reason : (
I removed it for the `animationEnd`:  b78ad24e8bc18ba8b8bb63e15934a6742af48433
and that worked.

Reply:
> It didn't pass for me for some reason : (
I removed it for the animationEnd: https://github.com/sonatype/sonatype-react-shared-components/commit/b78ad24e8bc18ba8b8bb63e15934a6742af48433
and that worked.

How about like this: fb8a3c55d